platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/conflicts/DefaultCapabilitiesConflictHandler.java

```diff
@@ -95,7 +96,16 @@ public boolean conflictExists() {
                         return true;
                     }
                 };
-                conflicts.add(new CapabilityConflict(group, name, candidatesForConflict));
+                // We now have a conflict, let's see if we already have one for this capability
+                capabilityIdToConflict.compute(capability.getCapabilityId(), (k, v) -> {
+                    if (v == null) {
+                        // Need to enqueue the conflict
+                        conflicts.add(k);
```

Comment:
💅 Perhaps I'm just used to using `compute` for thread-safe computations, but my usual assumption is that these mapping functions generally operate on their passed-in parameters and shouldn't mutate other state.

Reply:
I started with `compute` but since I don't have to merge values, a `put` and checking for `null` will be more than enough.